includes/Modules/AdSense.php

```diff
@@ -731,25 +685,90 @@ protected function parse_data_response( $method, $datapoint, $response ) {
 				case 'urlchannels':
 					// TODO: Parse this response to a regular array.
 					return $response->getItems();
-				case 'earning-today':
-				case 'earning-yesterday':
-				case 'earning-samedaylastweek':
-				case 'earning-7days':
-				case 'earning-prev7days':
-				case 'earning-this-month':
-				case 'earning-this-month-last-year':
-				case 'earning-28days':
-				case 'earning-prev28days':
-				case 'earning-daily-this-month':
-				case 'earnings-this-period':
-					// TODO: Parse this response to a regular array.
+				case 'earnings':
 					return $response;
 			}
 		}
 
 		return $response;
 	}
 
+	/**
+	 * Gets an array of dates for the given named date range.
+	 *
+	 * @param string $date_range Named date range.
+	 *                           E.g. 'last-28-days'.
+	 *
+	 * @return array|WP_Error Array of [startDate, endDate] or WP_Error if invalid named range.
+	 */
+	private function date_range_to_dates( $date_range ) {
+		switch ( $date_range ) {
+			case 'today':
+				return array(
+					date( 'Y-m-d', strtotime( 'today' ) ),
+					date( 'Y-m-d', strtotime( 'today' ) ),
+				);
+			case 'yesterday':
+				return array(
+					date( 'Y-m-d', strtotime( 'yesterday' ) ),
+					date( 'Y-m-d', strtotime( 'yesterday' ) ),
+				);
+			case 'same-day-last-week':
+				return array(
+					date( 'Y-m-d', strtotime( '7 days ago' ) ),
+					date( 'Y-m-d', strtotime( '7 days ago' ) ),
+				);
+			case '7-days':
+				return array(
+					date( 'Y-m-d', strtotime( '7 days ago' ) ),
+					date( 'Y-m-d', strtotime( 'yesterday' ) ),
+				);
+			case 'prev-7-days':
+				return array(
+					date( 'Y-m-d', strtotime( '14 days ago' ) ),
+					date( 'Y-m-d', strtotime( '8 days ago' ) ),
+				);
+			// Intentional fallthrough.
+			case 'daily-this-month':
+			case 'this-month':
+				return array(
+					date( 'Y-m-01' ),
+					date( 'Y-m-d', strtotime( 'today' ) ),
+				);
+			case 'this-month-last-year':
+				$last_year          = intval( date( 'Y' ) ) - 1;
+				$last_date_of_month = date( 't', strtotime( $last_year . '-' . date( 'm' ) . '-01' ) );
+
+				return array(
+					date( $last_year . '-m-01' ),
+					date( $last_year . '-m-' . $last_date_of_month ),
+				);
+			case '28-days':
```

Comment:
Same as above, this should instead become (and re-use) `last-28-days`.

Reply:
As above, this one uses the alternate end date. Ok to change?

Reply:
They should definitely become one (for both 7 and 28 days). There is no point in having two different versions. Whether we make them 7/28 days ago until yesterday or 6/27 days ago until today depends on whether the AdSense API even exposes data as recent as the last few hours. The only reason we don't include today in other modules is that there wouldn't be data anyway. @adamsilverstein Do you have any information about that?

In other words, it should either be `parse_date_range( $date_range, 1, 1 )` or `parse_date_range( $date_range, 1, 0 )`.